SC_redesign/Crop_and_Soil/soil/carbon_cycling/residue_partition.py

```diff
@@ -305,3 +305,281 @@ def _determine_plant_structural_carbon_amount(plant_dry_matter_residue_amount: f
             - plant_structural_to_slow_carbon_amount
 
         return updated_amount
+
+    @staticmethod
+    def _determine_soil_dry_matter_residue_amount(plant_dry_matter_residue_amount: float,
+                                                  tillage_fraction: float) -> float:
+        """Calculates the amount of soil dry matter residue at harvest
+
+        Parameters
+        ----------
+        plant_dry_matter_residue_amount: float
+            plant dry matter residue at harvest (kg/ha)
+        tillage_fraction: float
+            fraction of metabolic carbon incorporated into soil during tillage (unitless)
+
+        Returns
+        -------
+        float
+            the amount of soil dry matter residue at harvest (kg/ha)
+
+        References
+        -------
+        pseudocode_soil S.6.B.II.1
+
+        Notes
+        -------
+        This calculation does not exits in the original code, it's added in order for its usage in S.6.B.II.2
+        """
+        return plant_dry_matter_residue_amount * tillage_fraction
+
+    @staticmethod
+    def _determine_weighted_residue_dry_matter_lignin_fraction(soil_dry_matter_residue_amount: float,
+                                                               soil_biomass: float) -> float:
+        """Calculates the weighted fractional of lignin amount in residue dry matter
+
+        Parameters
+        ----------
+        soil_dry_matter_residue_amount: float
+            the amount of soil dry matter residue at harvest (kg/ha)
+        soil_biomass: float
+            below ground biomass (unknown)
+
+        Returns
+        -------
+        float
+            the weighted fractional of lignin amount in residue dry matter (unitless)
+
+        References
+        -------
+        pseudocode_soil S.6.B.II.2
+
+        Notes
+        -------
+        the referenced soil(below ground) biomass calculation was not found in the pseudocode_crop, neither is any
+        mention of biomass unit
+        """
+        # TODO: check with subject experts to determine biomass's unit isse #432
+        if soil_dry_matter_residue_amount + soil_biomass != 0:
+            return soil_dry_matter_residue_amount / (soil_dry_matter_residue_amount + soil_biomass)
+        else:
+            return 0
```

Comment:
I know I said you didn't need to change the variables, but I changed my mind - Sorry! This is more confusing than I thought. 
```suggestion
        if soil_dry_matter_residue_amount + root_biomass != 0:
            return soil_dry_matter_residue_amount / (soil_dry_matter_residue_amount + root_biomass)
        else:
            return 0
```